File: backend/src/main/java/com/github/fjbaldon/attendex/backend/dto/AttendanceSyncRequest.java
package com.github.fjbaldon.attendex.backend.dto;

import com.github.fjbaldon.attendex.backend.model.TimeSlotType;
import lombok.Data;

import java.time.Instant;
import java.util.List;

@Data
public class AttendanceSyncRequest {

    private List<Record> records;

    @Data
    public static class Record {
        private Long eventId;
        private Long attendeeId;
        private Instant checkInTimestamp;
        private TimeSlotType type;
    }
}

File: capture/app/src/main/java/com/github/fjbaldon/attendex/capture/data/remote/AttendanceSyncRequest.kt
package com.github.fjbaldon.attendex.capture.data.remote

import kotlinx.serialization.Serializable

@Serializable
data class AttendanceSyncRequest(
    val records: List<Record>
) {
    @Serializable
    data class Record(
        val eventId: Long,
        val attendeeId: Long,
        val checkInTimestamp: String
    )
}
